src/components/connectAdvanced.js

```diff
@@ -156,11 +156,10 @@ export default function connectAdvanced(
         if (this.selector.shouldComponentUpdate) this.forceUpdate()
       }
 
-      componentWillReceiveProps(nextProps) {
-        this.selector.run(nextProps)
-      }
-
-      shouldComponentUpdate() {
+      shouldComponentUpdate(nextProps) {
```

Comment:
Will try that. Is it called before sCU?

Reply:
Looks like it's not: https://reactjs.org/docs/react-component.html#updating

I can try to move `selector` from field on instance to state field, then it will be available in `static getDerivedStateFromProps`